### PR TITLE
build sim on CI

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: .github/workflows/test_and_build.yml [hunk 1]
```diff
@@ -152,16 +152,17 @@ jobs:
...
-  #   runs-on: ubuntu-latest
-  #   matrix:
-  #     state_machine: [accounting, test]
-  #   steps:
-  #     - uses: actions/checkout@v2
-  #     - run: ./scripts/install_zig.sh
-  #     - run: zig/zig run -OReleaseSafe -Dconfig-cluster-state-machine=${{matrix.state_machine}} src/simulator.zig -- 123
+  # It would be a good idea to also _run_ a single iteration,
+  # but that currently has some false failures :-)
+  simulator:
```
Comment (sentientwaffle) [inline .github/workflows/test_and_build.yml:168]: I think `simulator` needs to be added to `bors_tigerbeetle`'s dependencies.

Comment (matklad) [reply .github/workflows/test_and_build.yml:168]: Ah, that's annoying, thanks, fixed!

**Comment by matklad** [comment]
bors try

(there was a bug with .yml sytnax)

**Comment by matklad** [comment]
bors try

**Comment by matklad** [comment]
Finally sorted all build errors out!

bors r=sentientwaffle

File: src/test/state_machine.zig [hunk 2]
```diff
@@ -150,7 +150,7 @@ pub fn StateMachineType(comptime Storage: type, comptime constants_: struct {
             state_machine.grid.write_block(
                 next_tick_callback,
                 &state_machine.grid_write,
-                state_machine.grid_block,
+                &state_machine.grid_block,
```
Comment (matklad) [inline src/test/state_machine.zig:153]: Added `&` to placate the type checker, though I don't _fully_ understand the context here.

**Pull request merged**